packages/trie/src/trie/trie.ts

```diff
@@ -46,11 +56,29 @@ export class Trie {
     this._root = this.EMPTY_TRIE_ROOT
     this._deleteFromDB = opts?.deleteFromDB ?? false
 
+    if (opts?.db !== undefined) {
+      this._persistRoot = opts.persistRoot ?? true
+    } else {
+      this._persistRoot = false
+    }
+
     if (opts?.root) {
       this.root = opts.root
```

Comment:
```suggestion
      this.root = opts.root
      this.persistRoot()
```

I mean this.  That way, you don't do any every logic and just persist it the same way you do elsewhere in the class.